src/silly_sprites/utils.cpp

```diff
@@ -2,11 +2,29 @@
 #include <fstream>
 #include <sstream>
 
-[[nodiscard]] tl::optional<std::string> sly::read_file(std::filesystem::path const& path) {
-    auto file = std::ifstream{ path };
-    if (not file) {
-        return tl::nullopt;
+namespace sly {
+    static constexpr auto whitespace_chars = " \f\n\r\t\v";
+
+    [[nodiscard]] tl::optional<std::string> read_file(std::filesystem::path const& path) {
+        auto file = std::ifstream{ path };
+        if (not file) {
+            return tl::nullopt;
+        }
+
+        return (std::ostringstream{} << file.rdbuf()).str();
     }
 
-    return (std::ostringstream{} << file.rdbuf()).str();
-}
+    [[nodiscard]] std::string ltrim(std::string s) {
+        auto const pos = s.find_first_not_of(whitespace_chars);
+        s.erase(0, pos);
+        return s;
+    }
+    [[nodiscard]] std::string rtrim(std::string s) {
+        auto const pos = s.find_last_not_of(whitespace_chars);
+        s.erase(pos + 1);
+        return s;
+    }
```

Comment:
If you want, you can exchange this with an implementation that uses `std::isspace()` so we can get rid of the `whitespace_chars` constant. But please decide for yourself.
```suggestion
    [[nodiscard]] std::string ltrim(std::string s) {
        auto const it = std::find_if(
            std::cbegin(s),
            std::cend(s),
            [](char c) {
                return not std::isspace(static_cast<unsigned char>(c));
            });    
        s.erase(std::cbegin(s), it);
        return s;
    }
    
    [[nodiscard]] std::string rtrim(std::string s) {
        auto const it = std::find_if(
            std::crbegin(s),
            std::crend(s),
            [](char c) {
                return not std::isspace(static_cast<unsigned char>(c));
            }
        );
        s.erase(it.base(), std::cend(s));
        return s;
    }
```